bpf/bpf_lxc.c

```diff
@@ -999,9 +1000,21 @@ ipv6_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, __u8 *reason,
 	send_trace_notify6(ctx, TRACE_TO_LXC, src_label, SECLABEL, &orig_sip,
 			   LXC_ID, ifindex, *reason, monitor);
 
+#if !defined(ENABLE_ROUTING) && defined(ENCAP_IFINDEX) && !defined(ENABLE_NODEPORT)
+	/* In tunneling mode, we execute this code to send the packet from
+	 * cilium_vxlan to lxc*. If we're using kube-proxy, we don't want to use
+	 * redirect() because that would bypass conntrack and the reverse DNAT.
+	 * Thus, we send packets to the stack, but since they have the wrong
+	 * Ethernet addresses, we need to mark them as PACKET_HOST or the kernel
+	 * will drop them.
+	 * See #14646 for details.
+	 */
+	ctx_change_type(ctx, PACKET_HOST);
```

Comment:
This means that the packet will enter `ipvX_policy()` twice?

Reply:
Not as far as I can see. The packet will still leave for the stack. What makes you think it would be processed twice?

Reply:
My thinking was:

1. `ipvX_policy()` is invoked from bpf_overlay for a reply.
2. The reply is passed to stack, and the reverse DNAT translation is performed.
3. (This is where I can be wrong) The route lookup passes the reply to the veth which has bpf_lxc attached which executes `ipvX_policy()` again.

Reply:
Ah, I see. I don't think the current patch will change that. It's already happening without this patch: https://github.com/cilium/cilium/issues/14657.

I haven't had time to look at the above issue yet, but one of the two policy enforcements (from bpf_overlay or from bpf_lxc) likely should be compiled out and isn't.

Reply:
It's not important, but the current patch should change the path because `return redirect_ep(ifindex, from_host);` won't happen.

Reply:
@brb @pchaigno is the feedback here clear / addressed? Seems like this is the only open discussion topic for this 1.9-blocker.

Reply:
This was non-blocking comment from my side. 

> the current patch should change the path

I just want our understanding on this to be on the same page.

Reply:
Yes, the current patch will change the path of the packet to go through the stack instead of taking the `redirect()` shortcut. I do not think however that the packet path impacts #14657; the packet is already processed twice and will continue to be with this patch. We should fix #14657 in a separate PR.